project/frontend/src/components/PreferenceForm.js

```diff
@@ -64,27 +108,64 @@ class PreferenceForm extends React.Component {
           graduated
           progress
           onChange={(val) => this.changeWeightState(attrName, val)}
-          disabled={!Boolean(this.state[attrName]['pref'])}
+          disabled={!this.state[attrName]['pref']}
         />
       </div>
     );
   }
 
   handleSubmit(event) {
     event.preventDefault();
-    const { currLocation } = this.state;
-    currLocation['lat'] = this.props.currLocation.lat;
-    currLocation['lng'] = this.props.currLocation.lng;
-    this.setState({ currLocation });
-    console.log(this.state);
     this.props.history.push({
       pathname: '/match-results',
       state: this.state,
     });
   }
 
+  showCuisineOptions() {
+    const filter = createFilterOptions();
+    if (this.state.cuisineOptions) {
+      return (
+        <Autocomplete
+          multiple
+          name='cuisine'
+          id='cuisine'
+          options={this.state.cuisineOptions}
+          onChange={(_event, newCuisineList) => {
+            this.setState({ cuisine: newCuisineList });
+          }}
+          filterOptions={(options, params) => {
+            const filtered = filter(options, params);
+            const trimmedInput = params.inputValue.trim();
+            if (
+              trimmedInput !== '' &&
+              trimmedInput.length < 25 &&
+              // RegEx to make sure input is only chars and spaces.
+              /^[a-z\s]+$/i.test(trimmedInput) &&
+              // Make sure we don't display duplicates.
+              !(
+                options.includes(trimmedInput) &&
+                options.includes(params.inputValue)
+              )
```

Comment:
could we put this in a helper function isInputValid or something that checks all these things and change this to just if (isInputValid())